### DISCUSSION TITLE
How to Run Action that Failed Due to Outage?

### DISCUSSION THREAD

**Discussion by erikerikson** [community/community#49115] [Actions] [answered]
### Select Topic Area

Question

### Body

Yesterday's outage impacted one of my PRs with actions configured to run on push.  It has successfully run in its unchanged state reliably until yesterday.  There is no button or control to run the "ignored" workflow (see below).  I could manufacture a commit/commits to push but that is improper.  Is there a way to recover from your outage?

![image](https://user-images.githubusercontent.com/571200/222840870-c0320be1-7761-45f3-87cf-7e16d2ab3165.png)

**Comment by rauzerfabiano** [comment]
If your workflow failed due to the outage, and there is no button or control to run the ignored workflow, there are a few things you can try to recover:

    Re-run the workflow manually: You can manually trigger a new run of the workflow by going to the Actions tab in your repository and selecting the failed workflow run. From there, you can click on the “Re-run jobs” button to manually trigger a new run of the workflow.

    Use the API to trigger a new run: Another option is to use the GitHub API to trigger a new run of the workflow. You can do this by sending a POST request to the appropriate API endpoint with the necessary parameters. You can find more information on how to do this in the GitHub API documentation.

    Wait for the outage to be resolved: If the outage is still ongoing, it may be best to wait until it is resolved before attempting to run the workflow again. In the meantime, you can monitor the situation and keep an eye out for any updates or notifications from GitHub.

Overall, it's important to remember that outages are an unfortunate reality of working with any technology, and sometimes recovery may not be immediate or straightforward. However, by being proactive and exploring different options for recovery, you can minimize the impact of the outage on your work and get back on track as soon as possible.

**Reply by erikerikson** [reply]
This may be useful to people that land here from a search engine but aren't impacted by the issue I ask about.  The first option doesn't apply here as nothing failed (no jobs were run so that they could fail) and the third isn't relevant given that the outage was past both when I submitted the question and now when I am responding to you.  The second could be an answer and gives a little bread trail but doesn't enable.  As for the postlude, I felt like I was accepting of the outage and know how to work around it (i.e. 'I could manufacture a commit') but was looking for guidance from the team that created the product and caused the state I'm dealing with.  If in trying to help me they couldn't find an answer, it might indicate a gap in the product and thereby inform a need.

Regarding the second option, the API describes [creating a workflow dispatch event](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event) but my workflow doesn't trigger from one of those.  It also describes how to [interact with Workflow Runs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28) in there, there is a '[re-run a workflow](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#re-run-a-workflow)' option that may be relevant.  However, I'd need to set up a token and do other distracting work because the GitHub team failed to identify or simply address a failure state.  That's fine.  Some things are rare and aren't valuable enough to implement.  Sometimes we make assumptions that are wrong because we're all humans, even if augmented ;)  It still seems worthwhile to point out the product dead end and make space for a solution to be offered and in the meantime, workarounds.

**Reply by erikerikson** [reply]
Quick follow up...  The latest push has no workflow RUN_ID for me to use in the following ([from here](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#re-run-a-workflow)):

```
curl -L \
  -X POST \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/actions/runs/RUN_ID/rerun
  ```
  
  As a result, unless I am missing something, option two of your answer is not a solution.  Of course options one and two were immediately irrelevant.  Sorry to declare this starkly, I do so to save any scanning readers some time.

**Comment by jsoref** [comment, marked as answer]
File a ticket https://support.github.com/

I doubt it'll help, but the short answer is "you probably can't given your constraints".

If you expect outages to happen regularly, you could add enough workflow-dispatch support to all of your workflows so that you could manually trigger things.

Personally, I'm more likely to go w/ force-pushes to either rebase or just change the last commit's timestamp. If you need to not change an inbound PR's commits, you can add a commit that already exists in the destination branch as most merge strategies will effectively discard them (rebase directly, and merge indirectly).
